cli/command/service/opts.go

```diff
@@ -434,6 +436,11 @@ func newServiceOptions() *serviceOptions {
 func (opts *serviceOptions) ToService() (swarm.ServiceSpec, error) {
 	var service swarm.ServiceSpec
 
+	envVariables, err := runconfigopts.ReadKVStrings(opts.envFile.GetAll(), opts.env.GetAll())
```

Comment:
Thanks @thaJeztah. I updated the PR and duplicates of the env entries have been removed. I leave the order alone (not sorted), as that may potentially be different behavior from docker run. But let me know if we want to change both.